Lib/test/test_concurrent_futures.py

```diff
@@ -715,6 +715,10 @@ def test_default_workers(self):
         self.assertEqual(executor._max_workers,
                          (os.cpu_count() or 1) * 5)
 
+    def test_saturation(self):
+        list(self.executor.map(lambda x: mul(0, x), range(100 * self.executor._max_workers)))
+        self.assertEqual(len(self.executor._threads), self.executor._max_workers)
+
```

Comment:
Can you add a test launching one task at a time and checking that the number of threads doesn't grow?